shared/src/commonMain/kotlin/me/saket/press/shared/note/RealNoteRepository.kt

```diff
@@ -81,4 +82,13 @@ internal class RealNoteRepository(
       )
     }
   }
+
+  override fun markAsArchived(noteUuid: Uuid): Completable {
+    return completableFromFunction {
+      noteQueries.markAsArchived(
+          uuid = noteUuid,
+          archivedAt = clock.nowUtc()
+      )
+    }
+  }
```

Comment:
This is missing a test. I just pushed a test to master for checking `markAsDeleted()` that you can use as a reference.